### PR TITLE
HPCC-19028 Do not prevent ESP load due to DESDL configuration

### PR DESCRIPTION
**Pull request by rpastrana**
- Catches SmartSocket exception
- Logs clear indication of problem

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by rpastrana** [comment]
@mckellyln please review

File: esp/services/esdl_svc_engine/esdl_binding.cpp [hunk 1]
```diff
@@ -444,8 +444,22 @@ void EsdlServiceImpl::configureUrlMethod(const char *method, IPropertyTree &entr
 
     Owned<ISmartSocketFactory> sf = createSmartSocketFactory(iplist, true);
```
Comment (mckellyln) [inline esp/services/esdl_svc_engine/esdl_binding.cpp]: This line needs to be removed

File: esp/services/esdl_svc_engine/esdl_binding.cpp [hunk 2]
```diff
@@ -444,8 +444,22 @@ void EsdlServiceImpl::configureUrlMethod(const char *method, IPropertyTree &entr
...
 
-    connMap.remove(method);
-    connMap.setValue(method, sf.getClear());
+    try
+    {
+        Owned<ISmartSocketFactory> sf = createSmartSocketFactory(iplist, true);
```
Comment (mckellyln) [inline esp/services/esdl_svc_engine/esdl_binding.cpp]: Above lines use \"%s\" for method, perhaps match that here just to be consistent

File: esp/services/esdl_svc_engine/esdl_binding.cpp [hunk 3]
```diff
@@ -444,8 +444,22 @@ void EsdlServiceImpl::configureUrlMethod(const char *method, IPropertyTree &entr
...
+
+        connMap.remove(method);
+        connMap.setValue(method, sf.getClear());
+    }
+    catch(IException* ie)
+    {
+        ESPLOG(LogMin,"DESDL: Error while setting up connection for method '%s' verify its configuration.", method);
```
Comment (mckellyln) [inline esp/services/esdl_svc_engine/esdl_binding.cpp]: Above lines use "%s" for method, perhaps match that here just to be consistent

**Comment by mckellyln** [comment]
@rpastrana back at you

**Comment by rpastrana** [comment]
@mckellyln please re-review.

**Comment by mckellyln** [comment]
minor comment - could perhaps call
```
connMap.remove(method);
```
once beforehand then not both after successful createSmartSocketFactory() and in catch.

But really fine as is.
@richardkchapman ok to merge

**Review by mckellyln — APPROVED** [review]
Approved for merge

**Comment by rpastrana** [comment]
@richardkchapman please merge

**Comment by richardkchapman** [comment]
@rpastrana Please fix the typo in the commit message

**Comment by rpastrana** [comment]
@richardkchapman ready to merge

**Pull request merged**